### DISCUSSION TITLE
Question on example file(pytorch_geometric/examples/gcn.py)

### DISCUSSION THREAD

**Discussion by zehonyi21** [pyg-team/pytorch_geometric#9624] [Q&A]
I saw example gcn.py on 
'pytorch_geometric/examples/gcn.py'
(https://github.com/pyg-team/pytorch_geometric/blob/master/examples/gcn.py)
where the model code(line 57) is

def forward(self, x, edge_index, edge_weight=None):
        x = F.dropout(x, p=0.5, training=self.training)
        x = self.conv1(x, edge_index, edge_weight).relu()
        x = F.dropout(x, p=0.5, training=self.training)
        x = self.conv2(x, edge_index, edge_weight)
        return x

But is this supposed to be

def forward(self, x, edge_index, edge_weight=None):
        x = F.dropout(x, p=0.5, training=self.training)
        x = self.conv1(x, edge_index, edge_weight)
        x = F.relu(x)
        x = F.dropout(x, p=0.5, training=self.training)
        x = self.conv2(x, edge_index, edge_weight)
        return x

because the former one does not put x into relu()

Thank you for your opinion in advance

**Comment by EdisonLeeeee** [comment]
Hi @zehonyi21 

I don't see any differences between the two code snippets. Isn't it  in the first case that `relu` has been applied following `conv1`?

**Reply by zehonyi21** [reply]
I want to say that for the first case the output of 'self.conv1(x, edge_index, edge_weight)' is tensor by definition which means 'self.conv1(x, edge_index, edge_weight).relu()' is (tensor).relu(). However, pytorch says that it should be like 'relu(tensor)'
(https://pytorch.org/docs/stable/_modules/torch/nn/functional.html#relu)
Can you tell me what happens if we use (tensor).relu()?

**Reply by EdisonLeeeee** [reply]
I think `tensor.relu` should internally call `F.relu(tensor)` and they are essentially the same.

**Reply by zehonyi21** [reply]
Thank you that was what I wanted. Is there any reference link for that?

**Reply by EdisonLeeeee** [reply]
Sorry, there is no reference in the official pytorch doc.

**Reply by zehonyi21** [reply]
It's okay. If you have free time, can you put some comment on the code about this info? Since it seems unofficial, people might feel confused.
